confluent_kafka/avro/AvroProducer.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+#
+# Copyright 2016 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import logging
+
+from confluent_kafka.avro.serializer import SerializerError
+
+
+class AvroProducer(object):
+    '''
+        Kafka Producer client which does avro schema encoding to messages.
+        Handles schema registration, Message serialization.
+
+        Constructor takes below parameters
+
+        @:param: producer: confluent_kafka.Producer object
+        @:param: message_serializer: Message Serializer object
+    '''
+    def __init__(self, producer, message_serializer):  # real signature unknown; restored from __doc__
+        self.producer = producer
+        self.serializer = message_serializer
+        self.log = logging.getLogger(__name__)
+
+    def produce(self, topic, value=None, value_avro_schema=None, key=None, key_avro_schema=None, *args, **kwargs):
+        '''
+            Sends message to kafka by encoding with specified avro schema
+            @:param: topic: topic name
+            @:param: value: A dictionary object
+            @:param: value_avro_schema : Avro schema for value
```

Comment:
Yeah, people turn off compatibility for those cases. I find the pattern questionable, but people do it...

The most convenient way for users might be to just use `produce(*args, **kwargs)` and sort out the details in the header of the method (i.e. 1 positional arg = value, 2 position args = (key, value), kwargs just work by name). The obvious drawback is lack of documentation in the method signature.